src/Doctrine/DoctrineDataStore.php

```diff
@@ -51,8 +51,8 @@ public function getPaginated(string $type, int $page, FilterCollectionInterface
         } else {
             $pager = new Pagerfanta(new DoctrineORMAdapter($qb));
         }
-        $pager->setCurrentPage($page);
         $pager->setMaxPerPage($maxPerPage);
+        $pager->setCurrentPage($page);
```

Comment:
It's not possible to set the current page if it's not in sync with the max per page. Side issue fixed as well!